h2o-core/src/test/java/water/rapids/ast/prims/PermutationFeatureImportanceTest.java

```diff
@@ -0,0 +1,328 @@
+                    .withVecTypes(Vec.T_NUM, Vec.T_NUM)
+                    .withColNames("first", "second")
+                    .withDataForCol(0, ard(1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0))
+                    .withDataForCol(1, ard(1.1, 2.1, 3.1, 4.1, 5.1, 6.1, 7.1, 8.1))
+                    .build();
+            Scope.track(t);
+            Vec OG_vec = t.vec("first").doCopy();
+//            VecUtils.shuffleVec new_vv = new VecUtils.shuffleVec(t.vec(0)).doAll(t.vec("first"));
+            Vec new_v = Vec.makeVec(new VecUtils.shuffleVecVol2().doAll(t.vec("first"))._result, Vec.newKey());
+
+//            Vec new_v = new_vv._vec;
+            double rnd_coe = randomnessCoeefic(OG_vec, new_v);
+
+            Assert.assertNotEquals(0, rnd_coe);
+            
+        } finally {
+            Scope.exit();
+        }
+        
+    }
+
+    // compares values of vec and returns the randomness change. 0 no change, 1.0 all rows shuffled
+    double randomnessCoeefic(Vec og, Vec nw){
+        int changed_places = 0;
+        for (int i = 0; i < nw.length() ; ++i){
+            if(og.at(i) != nw.at(i))
+                changed_places++;
+        }
+        return changed_places * 1.0/nw.length();
+    }
+    
+    public double [] randomnessCoef(Frame fr, double [] res){
+        // copy vecs to check on permuated ones
+        Vec OG_vec[] = new Vec[fr.numCols()];
+        for (int i = 0 ; i <  fr.numCols(); i ++)
+            OG_vec[i] = fr.vec(i).makeCopy();
+
+        String [] features = fr.names();
+        ShuffleVec shf = new ShuffleVec(fr);
+        Vec shf_v;
+        for (int i = 0 ; i <  fr.numCols(); i ++){
+            String col_name = features[i];
+            shf_v = ShuffleVec.ShuffleTask.shuffle(fr.vec(col_name));
+            res[i] = randomnessCoeefic(OG_vec[i] , shf_v);
+        }
+        return res;
+    }
+    
+    @Test
+    public void shuffleVecBiggerData(){
+        Frame fr = null;
+        try {
+            Scope.enter(); 
+            fr = parse_test_file("./smalldata/airlines/AirlinesTrainWgt.csv"); // has 24422; should be enough to test speed 
+            Scope.track(fr);
+//            long timeElapsed_1 = endTime_using_set - startTime_using_set;
+            double res [] = new double [fr.numCols()];
+            res = randomnessCoef(fr, res);
+//            long timeElapsed_2 = endTime_copying_array - startTime_copying_array;
+            for (int i = 0 ; i < fr.numCols() ; ++i){
+                Assert.assertNotEquals(0, res[i]);
+                // TODO check if low random coeficent and try if shuffling again, if that doesnt work FIXME
+                // Its fine because the Vec can have the only a few values like Column[1] has only f10 and f1 
+            }
+            
+//            Assert.assertNotEquals(0, rnd_coe);
+        } finally {
+            Scope.exit();
+        }
+
+    }
+   
+    
+    @Test
+    public void Regression() {
+        GBMModel gbm = null;
```

Comment:
Maybe it is better to test results with the GLM model, which is more straightforward because it is base on linear regression.